### ISSUE TITLE
joeynmt 1.3 requires torchtext>=0.10.0, but you have torchtext 0.9.0 which is incompatible.

### ISSUE THREAD

**Issue by tanjie95** [joeynmt/joeynmt#158]
**torchtext conflict**
joeynmt 1.3 requires torchtext>=0.10.0, but you have torchtext 0.9.0 which is incompatible.

**To Reproduce**

B. From source
1.Clone this repository: git clone https://github.com/joeynmt/joeynmt.git
2.Install joeynmt and it's requirements: cd joeynmt pip3 install . (you might want to add --user for a local installation).
3.Run the unit tests: python3 -m unittest
Warning! When running on GPU you need to manually install the suitable PyTorch version (1.8.0) for your CUDA version. This is described in the PyTorch installation instructions.
Because the version of my cuda is 10.1. so I choose to install pytorch (1.8.0 cuda10.1)mannually. and then I install corresponding torchtext 0.9.0. Then I see the ERROR"joeynmt 1.3 requires torchtext>=0.10.0, but you have torchtext 0.9.0 which is incompatible." 

**Logged output**
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
joeynmt 1.3 requires torchtext>=0.10.0, but you have torchtext 0.9.0 which is incompatible.

**System (please complete the following information):**
 - OS: [e.g. iOS] linux 
 - CPU / GPU cuda 10.1
 - Python Version 3.7

**Comment by joeynmt** [comment]
@tanjie95 can you try again? The dependencies should now all be updated.

**Comment by HossamSayed99** [comment]
The issue still exists.

**Comment by may-** [comment]
Hi,

We are now working on getting rid of torchtext dependencies completely.
We test the new joeynmt without torchtext on torch1.11.0 + cuda11.5 + python3.9.

We aim to merge it to the main repo here soon! (Sorry, cannot promise a release date, yet. We still have a few problems to solve...)

**Comment by HossamSayed99** [comment]
Thank you for your response, is there a way to make the current release work? I tried cloning the repo and installing through pip both are broken. If I make changes in the exisitng code just to get things running and open a PR with them will it be considered?

**Comment by may-** [comment]
I only can think of rolling back the entire environment to the old one. i.e.
- python 3.7
- torch 1.9.1
- cuda 10.2
- torchtext 0.10.1

First, I separated the virtual env.
```bash
$ conda create -n py37 python=3.7
$ conda activate py37
(py37)$ python --version
Python 3.7.5
```

Re-install joeynmt in that environment:
```bash
(py37)$ pip install joeynmt
(py37)$ pip show joeynmt
Name: joeynmt
Version: 1.5.1
[...]
```

Downgrade torch and torchtext:
```
(py37)$ pip install --upgrade torch==1.9.1+cu102 -f https://download.pytorch.org/whl/torch_stable.html
(py37)$ pip install --upgrade torchtext==0.10.1
(py37)$ python -c "import torch; print(torch.__version__, torch.cuda.is_available())"
1.9.1+cu102 True
```

And I confirmed that the unittests passed.
```
(py37)$ python -m unittests
................................s.........................
----------------------------------------------------------------------
Ran 58 tests in 2.698s

OK (skipped=1)
```

I also tried to reproduce iwslt14 benckmark [here](https://github.com/joeynmt/joeynmt#iwslt14-de-en).
```
(py37)$ python -m joeynmt test transformer_iwslt14_deen_bpe/config.yaml 
2022-05-08 11:59:56,711 - INFO - root - Hello! This is Joey-NMT (version 1.5.1).
2022-05-08 11:59:56,712 - INFO - joeynmt.data - Building vocabulary...
2022-05-08 12:00:10,683 - INFO - joeynmt.data - Loading dev data...
2022-05-08 12:00:10,887 - INFO - joeynmt.data - Loading test data...
2022-05-08 12:00:11,087 - INFO - joeynmt.data - Data loaded.
2022-05-08 12:00:11,131 - INFO - joeynmt.prediction - Process device: cuda, n_gpu: 1, batch_size per device: 4096
2022-05-08 12:00:11,132 - INFO - joeynmt.prediction - Loading model from transformer_iwslt14_deen_bpe/best.ckpt
2022-05-08 12:00:18,867 - INFO - joeynmt.model - Building an encoder-decoder model...
2022-05-08 12:00:19,874 - INFO - joeynmt.model - Enc-dec model built.
2022-05-08 12:00:19,954 - INFO - joeynmt.prediction - Decoding on dev set (transformer_iwslt14_deen_bpe/valid.bpe.32000.en)...
2022-05-08 12:03:27,359 - WARNING - sacrebleu - That's 100 lines that end in a tokenized period ('.')
2022-05-08 12:03:27,360 - WARNING - sacrebleu - It looks like you forgot to detokenize your test data, which may hurt your score.
2022-05-08 12:03:27,360 - WARNING - sacrebleu - If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
2022-05-08 12:03:27,390 - INFO - joeynmt.prediction -  dev bleu[13a]:  34.52 [Beam search decoding with beam size = 5 and alpha = 1.0]
2022-05-08 12:03:27,390 - INFO - joeynmt.prediction - Decoding on test set (transformer_iwslt14_deen_bpe/test.bpe.32000.en)...
2022-05-08 12:06:23,733 - WARNING - sacrebleu - That's 100 lines that end in a tokenized period ('.')
2022-05-08 12:06:23,734 - WARNING - sacrebleu - It looks like you forgot to detokenize your test data, which may hurt your score.
2022-05-08 12:06:23,734 - WARNING - sacrebleu - If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
2022-05-08 12:06:23,762 - INFO - joeynmt.prediction - test bleu[13a]:  33.74 [Beam search decoding with beam size = 5 and alpha = 1.0]
```

**Comment by juliakreutzer** [comment]
This should be resolved with v2.0 and the upgrade to python 3.9, torch 1.11.
